pkg/json/encode.go

```diff
@@ -40,3 +40,27 @@ func (s Set) MarshalJSON() ([]byte, error) {
 	buf.WriteString(`]]`)
 	return buf.Bytes(), nil
 }
+
+func (cmr *CondMonitorParameters) UnmarshalJSON(p []byte) error {
+	var tmp []json.RawMessage
+	if err := json.Unmarshal(p, &tmp); err != nil {
+		return err
+	}
+	if err := json.Unmarshal(tmp[0], &cmr.DatabaseName); err != nil {
```

Comment:
I would generally rather use a counter:
i := 0
if err := json.Unmarshal(tmp[i], ..) {
}
i++
```
this makes things more flexible (and you never need to track count

Reply:
Done

Reply:
Yes, but all the time I put different fields for unmarshaling. In order to use a counter, I'd store the fields in some array, and grab them from there. 
I doubt that it will add readiness.

Reply:
ok